### PR TITLE
*: Add retry attempts into txn

### PR DESCRIPTION
**Pull request by shenli**
Set retry attempt number into txn, so the underlying storage engine could increase the priority of the txn which has larger retry attempt number.
@coocood @zimulala @c-wind

### CODE REVIEW

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
PTAL @coocood @zimulala

File: session_test.go [hunk 1]
```diff
@@ -2232,3 +2233,38 @@ func (s *testSessionSuite) TestSqlLogicTestCase(c *C) {
 	sql := "SELECT col0 FROM tab1 WHERE 71*22 >= col1"
 	mustExecMatch(c, se, sql, [][]interface{}{{"26"}})
 }
+
+func newSessionWithoutInit(c *C, store kv.Storage, dbName string) Session {
```
Comment (coocood) [inline session_test.go]: dbName is not used.

Comment (coocood) [inline session_test.go]: You can just return *session.

**Comment by coocood** [comment]
LGTM

File: kv/mock.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+// limitations under the License.
+
+package kv
+
+// mockTxn is a txn that returns a retryAble error when called Commit.
+type mockTxn struct {
+	opts map[Option]interface{}
+}
+
+// Always returns a retryable error
```
Comment (zimulala) [inline kv/mock.go]: add . at the end of a line.

File: kv/mock.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+func (t *mockTxn) Set(k Key, v []byte) error {
+	return nil
+}
+func (t *mockTxn) Delete(k Key) error {
+	return nil
+}
+
+// mockStorage is used to start a must commit-failed txn
```
Comment (zimulala) [inline kv/mock.go]: The same as this.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**